src/snowflake/snowpark/modin/plugin/_internal/ordered_dataframe.py

```diff
@@ -1119,6 +1130,7 @@ def join(
             "right",
             "inner",
             "outer",
+            "asof",
```

Comment:
It seems that when asof is used, the left_match_col, right_match_col and match comparator is required, i think we can add an assertion check here if how is asof, that  left_match_col, right_match_col and match comparator is not provided

Reply:
also, this condition here checks if the join can be skipped here. i don't think we can skip the asof join if the left on and right on is for row position column, because it depends on the match condition. however if match condition is on the row_position_column, and __ge__ and __lt__ is used as comparator, we could propbably skip the join here